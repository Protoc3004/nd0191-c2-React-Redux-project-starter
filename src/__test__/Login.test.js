import * as React from "react";
import { render, fireEvent } from "@testing-library/react";
import thunk from "redux-thunk";
import { within } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import Login from "../components/Login";

const mockStore = configureStore([thunk]);

describe("<Login />", () => {
  let store, Component;

  beforeEach(() => {
    store = mockStore({
      users: {
        sarahedo: {
          id: "sarahedo",
          name: "Sarah Edo",
        },
        tylermcginnis: {
          id: "tylermcginnis",
          name: "Tyler McGinnis",
        },
        johndoe: {
          id: "johndoe",
          name: "John Doe",
        },
      },
    });
  });

  it("will match Snapshot", () => {
    Component = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );

    expect(Component).toMatchSnapshot();
  });

  it("select a user", () => {
    const { getById } = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );

    Component = getById("Check Component");
    const selectOptions = within(Component).getAllByRole("option");
    const option = selectOptions[1];

    fireEvent.click(option);
    expect(option.getAttribute("aria-selected")).toBe("true");
  });
});
