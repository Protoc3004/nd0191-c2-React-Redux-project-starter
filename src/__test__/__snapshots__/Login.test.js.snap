// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login will match Snapshot 1`] = `
<div>
  <div
    class="container login-container"
  >
    <div
      class="d-flex justify-content-center h-100"
    >
      <div
        class="card card-login"
      >
        <div
          class="card-header text-white"
        >
          <h3>
            <strong>
              Sign In
            </strong>
          </h3>
        </div>
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group"
            >
              <label
                for="user"
              >
                Select a user
              </label>
              <select
                class="custom-select"
                name="user"
              >
                <option
                  value="sarahedo"
                >
                  Sarah Edo
                </option>
                <option
                  value="tylermcginnis"
                >
                  Tyler McGinnis
                </option>
                <option
                  value="zoshikanlu"
                >
                  Zenobia Oshikanlu
                </option>
              </select>
            </div>
            <button
              class="btn login_btn w-100"
              disabled=""
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
